addons/misra.py

```diff
@@ -3342,7 +3342,15 @@ def misra_config(self, data):
                     continue
                 if tok.astParent.str == "." and tok.astParent.valueType:
                     continue
-                self.report_config_error(tok, "Variable '%s' is unknown" % tok.str)
+                tok_pre = tok.previous
+                name = False
+                while tok_pre.isName or tok_pre.str == "*":
+                    if tok_pre.str != '*':
+                        name = True
```

Comment:
I believe we want to warn here:

     int x;
     if (x*unknown <= 30) {}

I think you should check if tok_pre is a keyword also.